### PR TITLE
Set the `close` method so that underlying files aggregated by NcML are closed

### LINKED ISSUE #41: Set `close` method to close all underlying files
### Description

`xncml.open_ncml` returns an `xr.Dataset` built from one or many netCDF files (children). Ideally, calling `close` on the returned Dataset would also close all the children.

### PR DESCRIPTION
**Pull request by huard**
- [x] Include documentation when adding new features.
- [x] Include new tests or update existing tests when applicable.

Set the `close` method to close all underlying opened files aggregated by NcML, similar to what `open_mfdataset` does.

Fixes #41

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,3 +4,4 @@ xarray
 cftime
 netCDF4
 dask
+psutil
```
Comment (aulemahal) [inline requirements.txt:7]: If this is only used in the tests, I don't think it should be added here, but rather in `ci/environment-dev-3.7.yml`.

**Review by aulemahal — APPROVED** [review]
As far as I understand file handling in xarray/netcdf, this looks good!

A small suggestion about the `psutil` requirement though.

File: requirements.txt (hunk 1 above)
Comment (huard) [reply requirements.txt]: Good catch, thanks !

Comment (huard) [reply requirements.txt]: Finally, no, the CI stuff seems to be not used. It's just github actions with the base dependencies.

**Pull request merged**